### DISCUSSION TITLE
Preventing duplicates?

### DISCUSSION THREAD

**Discussion by alexpdp7** [koxudaxi/datamodel-code-generator#1792] [Q&A]
Hi,

I'm generating models off <https://github.com/jgm/djot.js/blob/main/djot-schema.json>, and even with `--reuse-model`, I get a empty Pos2, Pos3, Pos4... empty extensions of the Pos type. Is there any way to fix this?

(There are also issues with tags, but I guess those are more difficult.)

Cheers,

Álex

**Comment by sayansisodiya** [comment]
Did you figure this out? Running into this as well

**Comment by nmoreaud** [comment]
Same here, many classes duplicated.

The tool doesn't seem to that two models that use the same reference are the same.

```yaml
user_a:
  $ref: "#/User"
  nullable: true
user_b:
  $ref: "#/User"
  nullable: true

# generates:
# class UserA()
# class UserB()
```

**Comment by koxudaxi** [comment]
Thank you for reporting the issue.
I just fixed it.
https://github.com/koxudaxi/datamodel-code-generator/pull/2903
